clients/ops/admin-ui/src/features/privacy-requests/SortRequestButton.tsx

```diff
@@ -0,0 +1,279 @@
+import { Flex, Icon, IconButton } from "@fidesui/react";
+import React, { useCallback, useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import {
+  clearSortFields,
+  selectPrivacyRequestFilters,
+  setSortDirection,
+  setSortField,
+} from "./privacy-requests.slice";
+
+type UseSortRequestButtonParams = {
+  sortField: string;
+  isLoading: boolean;
+};
+
+export enum ButtonState {
+  ASC = "asc",
+  DESC = "desc",
+  UNSELECTED = "unselected",
+}
+
+const useSortRequestButton = ({
+  sortField,
+  isLoading,
+}: UseSortRequestButtonParams) => {
+  const filters = useSelector(selectPrivacyRequestFilters);
+  const dispatch = useDispatch();
+  const [buttonState, setButtonState] = useState<ButtonState>(
+    ButtonState.UNSELECTED
+  );
+  const [wasButtonJustClicked, setWasButtonJustClicked] = useState(false);
+  useEffect(() => {
+    if (!isLoading) {
+      setWasButtonJustClicked(false);
+    }
+  }, [isLoading]);
+
+  useEffect(() => {
+    if (filters.sort_direction === undefined) {
+      setButtonState(ButtonState.UNSELECTED);
+    }
+  }, [filters]);
+
+  const handleButtonClick = useCallback(() => {
+    setWasButtonJustClicked(true);
+    dispatch(setSortField(sortField));
+    if (buttonState === ButtonState.UNSELECTED) {
+      dispatch(setSortDirection(ButtonState.ASC));
+      setButtonState(ButtonState.ASC);
+    }
+
+    if (buttonState === ButtonState.ASC) {
+      dispatch(setSortDirection(ButtonState.DESC));
+      setButtonState(ButtonState.DESC);
+    }
+
+    if (buttonState === ButtonState.DESC) {
+      dispatch(clearSortFields());
+      setButtonState(ButtonState.UNSELECTED);
+    }
+  }, [buttonState, setButtonState, dispatch, sortField]);
+
+  return {
+    handleButtonClick,
+    buttonState,
+    wasButtonJustClicked,
+  };
+};
+
+type ArrowIconProps = {
+  up?: boolean;
+};
+
+const ArrowIcon: React.FC<ArrowIconProps> = ({ up }) => {
+  if (up === undefined) {
+    return (
+      <Icon width="24px" height="26px" viewBox="0 0 24 26" fill="none">
```

Comment:
Should these embedded svg images be exported to clients/ops/admin-ui/src/app/features/common/Icon folder?

Reply:
Possibly. Normally our icons don't have any logic surrounding them. Since these 3 icons are tightly coupled together I put them in the `SortRequestButton.tsx` file. I think the it makes sense to move the `ArrowIcon` itself into that folder. If we need to refactor out the 3 sub svgs we can always do that later.